### PR TITLE
#1043 - Persistent Search Filters On Remount

### PR DESCRIPTION
**Pull request by hunyoboy**

### CODE REVIEW

File: src/plugins/curator/screens/search-teasers/delegate.js [hunk 1]
```diff
@@ -10,9 +10,13 @@ class Delegate extends AbstractDelegate {
   }
 
   getSearchParams() {
-    const { statuses } = this.component.props;
+    const { searchNodesRequestState: { request }, statuses } = this.component.props;
 
-    return { sort: SearchTeasersSort.ORDER_DATE_DESC.getValue(), statuses };
+    return {
+      sort: SearchTeasersSort.ORDER_DATE_DESC.getValue(),
```
Comment (gdbrown) [inline src/plugins/curator/screens/search-teasers/delegate.js]: @hunyoboy shouldn't sort come from the request too if request is populated?

File: src/plugins/dam/screens/search-assets/delegate.js [hunk 2]
```diff
@@ -7,6 +7,15 @@ class Delegate extends AbstractDelegate {
       schemas,
     }, dependencies);
   }
+
+  getSearchParams() {
+    const { searchNodesRequestState: { request }, statuses } = this.component.props;
+
+    return {
+      statuses,
```
Comment (gdbrown) [inline src/plugins/dam/screens/search-assets/delegate.js]: @hunyoboy what about sort here?

File: src/plugins/curator/screens/search-teasers/delegate.js (hunk 1 above)
Comment (hunyoboy) [reply src/plugins/curator/screens/search-teasers/delegate.js]: i agree.. thank you

File: src/plugins/dam/screens/search-assets/delegate.js (hunk 2 above)
Comment (hunyoboy) [reply src/plugins/dam/screens/search-assets/delegate.js]: @gdbrown i added the sort param

Comment (gdbrown) [reply src/plugins/dam/screens/search-assets/delegate.js]: @hunyoboy don't destruct enums, they are not plain ole objects

**Comment by hunyoboy** [comment]
> @hunyoboy are we certain we've covered all the search screens and the props/params that apply on each one?

As far as I know, yes.

**Comment by hunyoboy** [comment]
@gdbrown i believe i covered all the search forms that needed to be persistent. but please let me know if i miss something. thanks.

**Pull request merged**